clang/lib/CIR/CodeGen/CIRGenModule.cpp

```diff
@@ -449,7 +449,11 @@ static bool shouldAssumeDSOLocal(const CIRGenModule &CGM,
     return false;
 
   if (CGOpts.DirectAccessExternalData) {
-    llvm_unreachable("-fdirect-access-external-data not supported");
+    if (auto gv = dyn_cast<mlir::cir::GlobalOp>(GV.getOperation()))
```

Comment:
I think we've also traditionally copied over the comments from the original CodeGen, so that readers can get the full context without needing to refer to the original.